src/subtype.c

```diff
@@ -0,0 +1,569 @@
+// This file is a part of Julia. License is MIT: http://julialang.org/license
+
+/*
```

Comment:
Citation to a reference document would be very helpful. If your thesis is still accurate, you could use that.